test/framework/clusterctl/clusterctl_helpers.go

```diff
@@ -333,6 +334,7 @@ type ApplyCustomClusterTemplateAndWaitInput struct {
 	ClusterName                  string
```

Comment:
@CecileRobertMichon I could be mixing up some thing. But why are you moving from ApplyClusterTemplateAndWaitInput to ApplyCustomClusterTemplateAndWaitInput in CAPZ?

/hold

Reply:
xref: https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/3707/files#r1290886216
(Not sure if you're referring to this)

Reply:
Sorry, now understood

Reply:
Are you using the Flavor field?

Reply:
@sbueringer we're not moving from one to the other, we're implementing the `Waiter` interface https://github.com/kubernetes-sigs/cluster-api/blob/main/test/framework/clusterctl/clusterctl_helpers.go#L227 as part of `ControlPlaneWaiters`. I don't understand if you're saying there's another way around this?

Reply:
Okay took a closer look and saw that all starts with `EnsureControlPlaneInitialized` which takes the input struct and then calls `InstallCalicoAndCloudProviderAzureHelmChart` where the flavor is needed

Reply:
I think there are ways around it basically with closure, but looking at how you use the flavor it seems reasonable to require the flavor in this struct. So I would say let's add it